### PR TITLE
bochs: Added support for customizing debugger

### PR DESCRIPTION
**Pull request by ibalajiarun**
--with-gdb-stub will enable GDB debugger
--with-debugger will enable generic debugger

### CODE REVIEW

File: bochs.rb [hunk 1]
```diff
@@ -13,11 +13,20 @@ class Bochs < Formula
...
+    opts << "--enable-gdb-stub"
+  else
+    opts << "--enable-debugger"
+  end
 
   def install
     system "./configure", "--prefix=#{prefix}",
                           "--with-x11",
-                          "--enable-debugger",
+                          *opts,
```
Comment (dunn) [inline bochs.rb]: Ruby 1.8 on Mountain Lion can't deal with a splat in the middle of a list, it'll need to be moved to the end.

File: bochs.rb [hunk 2]
```diff
@@ -13,11 +13,20 @@ class Bochs < Formula
   depends_on "pkg-config" => :build
   depends_on :x11
   depends_on "gtk+"
+  
+  option "with-gdb-stub", "Enable GDB Stub"
+  
+  opts = []
```
Comment (DomT4) [inline bochs.rb]: This is in the wrong place, you need to pass arguments inside `def install`, not outside of it. Let's also not create an empty array just for one option.

Do something like this:

```ruby
def install
  args = %W[
    --prefix=#{prefix}
    --with-x11
    --enable-disasm
    --disable-docbook
    --etc
    --etc
    --etc
  ]

  if build.with? "gdb-stub"
    args << "--enable-gdb-stub"
  else
    args << "--enable-debugger"
  end
```

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

**Comment by apjanke** [comment]
@BrewTestBot test this please.

**Comment by apjanke** [comment]
Merged. Thank you for your contribution to Homebrew!

**Pull request closed without merging**